src/main/kotlin/calculator/InputAnalyzer.kt

```diff
@@ -0,0 +1,47 @@
+package calculator
+
+class InputAnalyzer(private var inputString: String) {
+    private val delimiters: MutableList<String> = DEFAULT_DELIMITER.toMutableList()
+
+    init {
+        extractCustomDelimiter()
+    }
+
+    private fun extractCustomDelimiter() {
+        val delimiterMatch = CUSTOM_DELIMITER_PATTERN.find(inputString)
+
+        delimiterMatch?.let {
+            delimiters.add(it.groupValues[1])
+            this.inputString = it.groupValues[2]
+        }
+    }
+
+    fun extractNumbers(): List<Int> {
+        return inputString.split(*delimiters.toTypedArray())
+            .map { it.toInt() }
+    }
+
+    fun validateInput() {
+        inputString.split(*delimiters.toTypedArray()).forEach { token ->
```

Comment:
사소한 부분이지만 `Array`로 변경하는 것도 좋지만 정규식으로 변경해서 분리하는 방법도 괜찮은 것 같아요. 🤔
```suggestion
        val regex = delimiters.joinToString("|").toRegex()
        inputString.split(regex).forEach { token ->
```

Reply:
와우 "|"로 List를 하나로 묶어서 정규표현식으로 만드는 아주 좋은 방법이군요!
감사합니다 👍